File: src/models/person.go
package models

import (
	"errors"
	"lucree/src/service"
)

type Person struct {
	First_name string `json:"first_name,omitempty"`
	Last_name  string `json:"last_name,omitempty"`
	Birthday   string `json:"birthday,omitempty"`
	Password   string `json:"password,omitempty"`
	Username   string `json:"username,omitempty"`
	User_id    int64  `json:"user_id,omitempty"`
}

func (person *Person) Prepare() error {
	if err := person.validate(); err != nil {
		return err
	}

	passHash, err := service.Hash(person.Password)

	if err != nil {
		return err
	}

	person.Password = string(passHash)
	return nil
}
func (person *Person) validate() error {
	if person.First_name == "" || person.Last_name == "" || person.Birthday == "" || person.Password == "" || person.Username == "" {
		return errors.New("All fields must be required")
	}

	return nil
}

File: src/models/person.py
from flask_sqlalchemy import SQLAlchemy
from src.config.database import db
from hashlib import sha256

class Person(db.Model):
    id = db.Column(db.Integer(), primary_key=True)
    first_name = db.Column(db.String(255))
    last_name = db.Column(db.String(255))
    birthday = db.Column(db.String(255))
    username = db.Column(db.String(255))
    password = db.Column(db.String(255))

    def crypto_password(self):
        self.password = sha256(self.password.encode('utf-8')).hexdigest()
    
    def verify_password(self, password):
        return sha256(password.encode('utf-8')).hexdigest() == self.password

    def to_json(self):
        return {
            "id": self.id,
            "first_name": self.first_name,
            "last_name": self.last_name,
            "birthday": self.birthday,
            "username": self.username,
            "password": self.password
        }